Dark sub navigation fixes

bartaz opened the pull request:
## Done

Fixes #2752 
Fixes #2753 

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/ or demo
- Go to [dark subnavigation example](https://vanilla-framework-canonical-web-and-design-pr-2761.run.demo.haus/examples/patterns/navigation/subnav-dark/)
- Make sure subnav background is the same as main nav
- Make sure there is no light border around dark subnav dropdown
- On small screen there should be no additional indent and no border around subnav items


## Screenshots

<img width="596" alt="Screenshot 2020-01-20 at 16 49 48" src="https://user-images.githubusercontent.com/83575/72739905-ea4fe000-3ba4-11ea-82d2-3af04ae49a33.png">
<img width="702" alt="Screenshot 2020-01-20 at 16 49 33" src="https://user-images.githubusercontent.com/83575/72739934-f89dfc00-3ba4-11ea-820e-a81353bc0ec6.png">

webteam-app commented:
Starting demo at: https://vanilla-framework-canonical-web-and-design-pr-2761.run.demo.haus/

lyubomir-popov commented:
@bartaz may I suggest a couple of driveby's:

- For the dark examples, the logo has black text on dark grey background, I've made a version with white text:
https://assets.ubuntu.com/v1/20673d9f-lxd_primary--for-dark-bg.svg
- It would be nice if the subnav items could use the default vf-focus focus state (currently they use the browser default dotted version)

lyubomir-popov commented on scss/_patterns_subnav.scss:
```diff
@@ -50,11 +50,17 @@
 
+      .p-subnav__items,
+      .p-subnav__items--right {
+        @extend %vf-is-bordered;
```
Not sure %vf-is-bordered is a good idea. The goal of the theming is to set only colours, keeping things like border-width in the rules defined above. This way, it wouldn't need to override the colour set in this placeholder.

lyubomir-popov replied:
besides, the colours of these lines for both themese should already be set by `.p-navigation.is-dark .p-navigation__link>a::before` defined in p-navigation. So not sure why the borders are set here at all?

bartaz replied:
@lyubomir-popov 
I think the current implementations is so confusing because we have border around subnav in light theme, but don't have the border in the dark theme. This causes issues when `is-dark` is applied and border is not removed.

I'll try to change it, so that the border is used in dark theme, but has same colour as background. To make it consistent and easier to theme.

lyubomir-popov commented on scss/_patterns_subnav.scss:
```diff
@@ -73,11 +79,17 @@
 
+      .p-subnav__items,
+      .p-subnav__items--right {
+        border: 0;
```
if the rule at row 61 didn't set the borders, this would also be unnecessary

lyubomir-popov commented:
@bartaz I expereimented removing some more stuff that I think isn't needed in this [branch](https://github.com/lyubomir-popov/vanilla-framework/tree/subnav-further-fixes).

Also, there's a problem with chevron colour - it should the variables I've added in the branch above from the theme colours. But it doesn't work.

bartaz commented:
@lyubomir-popov I updated this PR.

I removed %vf-is-bordered from theme code replacing it with changing just the color of the border. For consistency dark theme now also has border (previously only light had).

I also updated the chevron colors as in your branch, it seems to work for me (percy detected the change of chevron color).

I updated the logo in dark example - thanks for the image.

As for the focus - I left it as it is for now. Both p-navigation and p-subnav now use default browser focus consistently, if we want to change it we should do it in both places probably. Let's do it separately from this.

lyubomir-popov commented:
Thanks @bartaz looks much better, adding +1. 

One last question - and this is completely optional, but why not simply delete the code in red?

<img width="1014" alt="Screenshot 2020-01-21 at 09 25 41" src="https://user-images.githubusercontent.com/2741678/72791971-1c5a5400-3c30-11ea-8d84-312d04c1ef57.png">


The only borders we need are the pseudo elements in the screenshot below, and they are added in the p-navigation file:

<img width="1010" alt="Screenshot 2020-01-21 at 09 32 54" src="https://user-images.githubusercontent.com/2741678/72792553-216bd300-3c31-11ea-8a61-d086df025216.png">

bartaz commented:
@lyubomir-popov Created https://github.com/canonical-web-and-design/vanilla-framework/issues/2765 if we want to get rid of the border. Feel free to comment there.
